server/xpub-server/routes.js

```diff
@@ -16,37 +13,13 @@ const nocache = (req, res, next) => {
   next()
 }
 
-const s3 = new AWS.S3({
-  ...config.aws.credentials,
-  ...config.aws.s3,
-  apiVersion: S3_API_VERSION,
-})
-
 let lastHealthStatus = null
```

Comment:
I notice this `lastHealthStatus` is used to try to avoid too many `error` logs. I think that should not happen at all, because if the healthcheck fails (several times in a row) the node is disconnected and the load balancer stops sending traffic to it.

Reply:
So could probably be "simplified out"

Reply:
So should the app shutdown rather than attempt to recover itself?

Reply:
No, I'm saying it's not needed to shut down nor to track the previous health status; `/ping` can be stateless and the app can always keep running; there may be an external fix like stopping/starting the RDS database that will make it serve requests again.

Reply:
... so in that example it would get added back into the `TargetGroup` of the LB? ... I've just worked with too many systems that can flood the logs in failure cases - just want to ensure we never do. Would you be happy if it were to go in?

Reply:
The ping check would keep being requested even if its out of the load balancer, to see if the node has recovered. Can you confirm @giorgiosironi?

Reply:
Yes I think you're right. In our typical experience we turn off nginx during deploy, which explains why we do not see logs as the request is refused at the TCP level rather than being answered with a HTTP 500. So this part should be unchanged.